OBW fallback: Replace WooCommerce Shipping suggestion

kallehauge opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

This PR aims to update our fallback onboarding wizard experience by replacing "WooCommerce Shipping & Tax" (slug: `woocommerce-services`) with "WooCommerce Shipping" (slug: `woocommerce-shipping`) as the suggested shipping plugin for US based stores.

Closes #51861

**Other notes:**

1. We're also updating the shipping copy to include `UPS` as a carrier to reflect the supported carriers in the new shipping plugin.
2. WCS&T and WC Shipping have "parallel support", so although both includes shipping in their name (for now), then they can coexist and WCS&T will only use its tax feature.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Spin up this branch on a fresh installation
2. Set the `woocommerce_show_marketplace_suggestions` option to `no` (`wp option set woocommerce_show_marketplace_suggestions no`)
3. Go to WooCommerce > Home and you should be redirected to the onboarding wizard (`/wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard&step=plugins`)
4. Select any store description/categorisation to get to the next step
5. The only store information we care about on the second step is `US` since it's a requirement for WC Shipping to show up
6. Go to the next step and you should now see a list of suggested plugins
7. Verify: that the shipping plugin description includes `UPS`
8. Verify: that "WooCommerce Shipping" is included in the terms of service copy at the bottom of the page.
9. Verify: that "WooCommerce Tax" is included in the terms of service copy at the bottom of the page.
10. Open the plugins page in a new window (`/wp-admin/plugins.php`)
11. Add and activate `WooCommerce Shipping` (you can search for it in WP Admin)
12. Go back to the onboarding wizard and refresh the page
13. Verify that the shipping plugin has an `Installed` label
14. Go back to the plugins page (`/wp-admin/plugins.php`)
15. Deactivate and delete WooCommerce Shipping
16. Add and activate `WooCommerce Shipping & Tax`
17. Go back to the onboarding wizard and refresh the page
18. Verify that the WooCommerce Tax plugin has an `Installed` label
19. Go back to the plugins page (`/wp-admin/plugins.php`)
20. Deactivate and delete WooCommerce Shipping & Tax
21. Go back to the onboarding wizard and refresh the page
22. Make sure both the Shipping and Tax plugins are selected and continue the onboarding wizard
23. _If you see the "Connect your account" step, then it's OK to skip it in the top right corner_
24. Go to the plugins page (`/wp-admin/plugins.php`)
25. Verify that both `WooCommerce Shipping` and `WooCommerce Shipping & Tax` are installed and activated


<!-- End testing instructions -->

### Screenshots


**The OBW step in question**

![Screenshot 2025-02-05 at 21 19 22](https://github.com/user-attachments/assets/340eb553-e5ef-41c1-9308-947b1be41984)


### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

kallehauge committed:
OBW: Replace WooCommerce Shipping suggestion

kallehauge committed:
Fix linting

kallehauge committed:
Update DefaultFreeExtensionsTest test suite

kallehauge committed:
Make tests more specific to avoid false positives

kallehauge committed:
Revert changes to unrelated test file

kallehauge committed:
Revert unrelated data package change

kallehauge committed:
Revert test changes for unrelated data package change

kallehauge committed:
Add changelog entry

kallehauge committed:
Merge branch 'trunk' into update/51861-update-obw-shipping-recommendation

kallehauge committed:
Merge branch 'trunk' into update/51861-update-obw-shipping-recommendation

harriswong commented:
Waclaw is AFK. I will take a look at this. 👀

harriswong commented on plugins/woocommerce/src/Internal/Admin/RemoteFreeExtensions/DefaultFreeExtensions.php:
```diff
@@ -292,7 +293,7 @@ public static function get_plugin( $slug ) {
 				'min_wp_version' => '5.9',
 			),
-			'woocommerce-services:shipping' => array(
+			'woocommerce-shipping'          => array(
```
The conditions here makes sense to me. It will show if the country is in US and `woocommerce-shipping` is not activated.

Question: This may not be related to the PR but do you know what the `dot_notation` in the `or` block is for please?

kallehauge replied:
> This may not be related to the PR but do you know what the dot_notation in the or block is for please?

AFAIK then "product_types" is part of the old/legacy OBW where you select which type of products you want to sell. If you select "only downloads" (it might be called "virtual products" or something similar - can't remember it specifically) then we hide WC Shipping as a recommendation since we only ship physical products 😄 

I feel decently sure we don't need it anymore and it's just being ignored for the new OBW because we use `or` but, at the same time, the legacy/old OBW is still part of the WC Core code, so while that still - theoretically - exists and can be shown if the `core-profiler` feature flag is removed then I think we should keep it around. I won't die on a hill about it though, since I don't we ever intend on showing the old profiler/OBW anymore 😛 

_And "dot notation" is a way of traversing the value being passed (think `explode( '.' )`)_

harriswong replied:
+1 on keeping it.

harriswong commented on plugins/woocommerce/src/Internal/Admin/RemoteFreeExtensions/DefaultFreeExtensions.php:
```diff
@@ -307,15 +308,6 @@ public static function get_plugin( $slug ) {
-						),
-					),
 					array(
 						'type'    => 'not',
```
Do we need the "is visible if woocommerce-shipping is not activated" condition? The other plugins don't seem like they need to declare that if the plugin is themselves. 

If we don't need it, then we can delete all of them to simplify the conditions.

kallehauge replied:
Nope! They weren't necessary 😄 I've removed them from WCS&T and WC Shipping 👏 
The condition still exist for two other plugins (Jetpack and Google for WooCommerce) but I don't want to modify them in this PR since it's a bit outside scope and I don't want to risk something if it's used later by some other hook for the specific plugins 🙈

harriswong reviewed (approved):
Thanks for the PR! 👍  LGTM! I left some comments but not a blocker.

> Verify: that "WooCommerce Shipping" is included in the terms of service copy at the bottom of the page.
> Verify: that "WooCommerce Tax" is included in the terms of service copy at the bottom of the page.

"WooCommerce Shipping" isn't on the ToS, but "Woo Shipping" is. I think that's ok. Same for "Woo Tax".

✅ Verified the onboarding after installation. 
![image](https://github.com/user-attachments/assets/84c08835-8b41-4626-b13e-21d49daa0db7)

kallehauge commented:
> >Verify: that "WooCommerce Shipping" is included in the terms of service copy at the bottom of the page.
> >Verify: that "WooCommerce Tax" is included in the terms of service copy at the bottom of the page.
>
>"WooCommerce Shipping" isn't on the ToS, but "Woo Shipping" is. I think that's ok. Same for "Woo Tax".

@harriswong verifying we're looking at the same thing 😛 I remember the "Woo Shipping" naming from the [WPCOM ToS](https://wordpress.com/tos/) page (thank you for checking as well!) - this PR should make it so the `WooCommerce Shipping` plugin is shown in the ToS paragraph at the bottom of the OBW page:

![410178866-340eb553-e5ef-41c1-9308-947b1be41984](https://github.com/user-attachments/assets/04baaed4-c62a-4778-b635-e75f58f0d065)

harriswong commented:
Sorry, yes! The text under the "Continue" button is fine. It was the "Terms of Service" `<a href>` link that says "Woo Shipping". 👍

kallehauge committed:
Merge branch 'trunk' into update/51861-update-obw-shipping-recommendation

kallehauge committed:
Remove redundant plugin activated condition

kallehauge committed:
Remove redundant WCS&T plugin activated condition

kallehauge committed:
Merge branch 'trunk' into update/51861-update-obw-shipping-recommendation

kallehauge committed:
Merge branch 'trunk' into update/51861-update-obw-shipping-recommendation

kallehauge committed:
Remove outdated test

We have removed the plugins_activated condition for WCS&T so we expect it to be recommended - the difference being that it's shown as "installed".
This matches what the other plugins in the default recommendations are doing.

kallehauge committed:
Remove redundant activation test

This is already done by the following test in combination:
* test_wcshipping_is_recommended_for_shipping
* test_wcservices_is_recommended_for_tax
